twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -314,6 +318,31 @@ scrooge_scala_library = rule(
     provides = [DefaultInfo, ScroogeInfo, JavaInfo],
 )
 
+def _create_scala_struct(ctx):
+  """Create a scala provider in the shape expected by the IntelliJ bazel plugin."""
```

Comment:
I guess we are transitively putting things here?


This part is confusing to me because it seems some jars will end up in many targets.

Why doesn't this work to just put `class_jar` from the current target? Why do we need the transitive part?

Reply:
It's transitive in the sense that it pulls jars from the scrooge_scala_library's deps. Is that what you mean? I'm not sure I understand the question. Let me explain my understanding and see if that clears it up or if I'm mistaken:

My understanding is, suppose we have some scrooge target:
```
scrooge_scala_library(
    name = "my_scrooge",
    deps = [":thrift1", ":thrift2"]
)
```

The aspect will traverse the deps, call scrooge against each target's sources, produce a jar for each, and store a reference to those jars in each dep's `ScroogeAspectInfo.java_info`. So the result is `:thrift1` and `:thrift2` have jars that I need to return to IJ stored in their `ScroogeAspectInfo.java_info.outputs`.

This function gathers up all those jars by iterating through `:thrift1` and `:thrift2`, pulls the jars out of `ScroogeAspectInfo.java_info.outputs` and puts them in a place IJ can find them.

It also gets the jars for anything a scrooge_scala_library exports since those symbols might be used in the editor.